### PR TITLE
git: use forward slashes in relative path to backing git repo

### LINKED ISSUE #2403: Repositories created in Windows cannot be used within WSL
## Description

When creating a repository in Windows, the file `.jj/repo/store/git_target` contains backslash `\` as the path separator. If this jj repository is accessed from within WSL, operations fail because jj is expecting forward slashes.

It seems to still work in Windows if `git_target` contains forward slashes, so perhaps jj should always write forward slashes there, at least if it's a relative path (if jj ever writes an absolute path like `\\?\C:\...`, that probably won't work in WSL anyway so it wouldn't matter).

## Steps to Reproduce the Problem

Here's an example terminal session demonstrating the issue.

```
# CREATE A NEW COLOCATED REPOSITORY

PS> git init slashes-test
Initialized empty Git repository in C:/src/slashes-test/.git/

PS> cd slashes-test

PS> echo "hello" > readme.md

PS> git add -A; git commit -m 'initial commit'
[trunk (root-commit) e7356d5] initial commit
 1 file changed, 1 insertion(+)
 create mode 100644 readme.md

PS> jj init --git-repo=.
Done importing changes from the underlying Git repo.
Initialized repo in "."

> jj
@  olvwtvyn (no email set) 18 seconds ago 32250eb6
│  (no description set)
◉  zrwrnyxq jennings@example.com 29 seconds ago trunk HEAD@git e7356d53
│  initial commit
◉  zzzzzzzz root() 00000000

# TRY TO READ IT FROM WSL

> wsl
$ jj
Internal error: The repository appears broken or inaccessible: Cannot access /mnt/c/src/slashes-test/.jj/repo/store/..\..\..\.git

$ cat .jj/repo/store/git_target
..\..\..\.git$

# REPLACE THE git_target CONTENTS WITH FORWARD SLASHES

$ echo -n "../../../.git" > .jj/repo/store/git_target
$ jj
@  olvwtvyn (no email set) 2023-10-19 15:14:15.000 -07:00 b0df79f7
│  (no description set)
◉  zrwrnyxq jennings@example.com 2023-10-19 15:12:43.000 -07:00 trunk HEAD@git e7356d53
│  initial commit
◉  zzzzzzzz root() 00000000


# VERIFY IT STILL WORKS IN WINDOWS

$ exit
logout

PS> jj
@  olvwtvyn (no email set) 6 seconds ago b0df79f7
│  (no description set)
◉  zrwrnyxq jennings@example.com 1 minute ago trunk HEAD@git e7356d53
│  initial commit
◉  zzzzzzzz root() 00000000

```

## Expected Behavior

`jj log` should succeed.

## Actual Behavior

```
$ jj
Internal error: The repository appears broken or inaccessible: Cannot access /mnt/c/src/slashes-test/.jj
```

## Specifications

- Platform: Windows 11
- Version: jj 0.10.0-eee140eef3f711f5114684aae39025b1c7e5d071

### PR DESCRIPTION
**Pull request by martinvonz**
Closes #2403.

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by jennings** [comment]
This fixes the issue on my machine, thank you!

```
PS> jj --version
jj 0.10.0-f714afefc13a0b0994027f97b53c1ba2fdfc325a

PS> git init slashes-test
Initialized empty Git repository in C:/src/slashes-test/.git/

PS> cd slashes-test

PS> echo "hello" > readme.md

PS> git add -A; git commit -m 'initial commit'
[trunk (root-commit) 7a93317] initial commit
 1 file changed, 1 insertion(+)
 create mode 100644 readme.md

PS> jj init --git-repo=.
Done importing changes from the underlying Git repo.
Initialized repo in "."

PS> cat .\.jj\repo\store\git_target
../../../.git

PS> jj
@  zlytyrrn (no email set) 2 seconds ago 4ef1abe4
│  (no description set)
◉  nquxkszz jennings@example.com 7 seconds ago trunk HEAD@git 7a933173
│  initial commit
◉  zzzzzzzz root() 00000000

PS> wsl
$ jj
@  zlytyrrn (no email set) 2023-10-19 19:34:25.000 -07:00 683fa659
│  (no description set)
◉  nquxkszz jennings@example.com 2023-10-19 22:44:03.000 -07:00 trunk HEAD@git 7a933173
│  initial commit
◉  zzzzzzzz root() 00000000

$ exit
logout

PS> jj
@  zlytyrrn (no email set) 3 hours ago 683fa659
│  (no description set)
◉  nquxkszz jennings@example.com 36 seconds ago trunk HEAD@git 7a933173
│  initial commit
◉  zzzzzzzz root() 00000000
```

File: lib/src/git_backend.rs [hunk 1]
```diff
@@ -138,8 +138,20 @@ impl GitBackend {
         let extra_path = store_path.join("extra");
         fs::create_dir(&extra_path).context(&extra_path)?;
         let target_path = store_path.join("git_target");
-        fs::write(&target_path, git_repo_path.to_str().unwrap().as_bytes())
-            .context(&target_path)?;
+        if cfg!(windows) && git_repo_path.is_relative() {
+            // WSL doesn't like backslashes in paths, so we manually format the path with
+            // forward slsashes (non-WSL is fine with forward slashes). If the path is
+            // absolute, there's much we can do - it simply won't work on both inside and
+            // outside WSL.
```
Comment (thoughtpolice) [inline lib/src/git_backend.rs:145]: Typo on `slsashes`, but I'd probably just reword this to be a bit more clear about the exact condition under which an error occurs:

```
// When a repository is created in Windows, format the path with *forward slashes*
// and not backwards slashes. This makes it possible to use the same repository under
// Windows Subsystem for Linux.
```

**Review by thoughtpolice — CHANGES_REQUESTED** [review]

File: lib/src/git_backend.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/git_backend.rs]: Done (with a few additional typo fixes)

**Review by thoughtpolice — APPROVED** [review]
